openwisp_monitoring/monitoring/base/models.py

```diff
@@ -766,7 +792,15 @@ def _is_crossed_by(self, current_value, time=None, retention_policy=None):
                     continue
                 utc_time = utc.localize(datetime.utcfromtimestamp(point['time']))
                 # did this point cross the threshold? Append to result list
-                results.append(self._value_crossed(point[self.metric.field_name]))
+                # check if alert_on_related_field is present in metric configuration
+                if 'alert_on_related_field' in self.metric.config_dict:
+                    results.append(
+                        self._value_crossed(
+                            point[self.metric.config_dict['alert_on_related_field']]
+                        )
+                    )
+                else:
+                    results.append(self._value_crossed(point[self.metric.field_name]))
```

Comment:
isn't the final return statment suppposed to be updated here??

```
return self._time_crossed(time) and value_crossed
```

Is not taking in consideration the related field.

What if value_crossed is False but the related value is crossed?

What about `if self.tolerance == 0:`???

So now I have some more basic quesitons:

1. Why was this kind of change, which is very delicate, introduced in this huge PR? Do we have to introduce this change or can we make the iperf check work without it for now? 

2. Looking at this code, I am pretty scared to merge it! I do not want to do this.
@Aryamanz29 can you extract these commits into a separate branch please? I want to merge the rest of the work but not this right now, I think it needs more careful thought.

Reply:
1. Yes, we can make `iperf check` work without this feature (check will run without any `alertsettings`)
2. I'll remove `alert_on_related_field` feature from this PR as of now & will open a separate PR :+1: